spec/source_map/clip_spec.rb

```diff
@@ -1052,6 +1052,25 @@ def foo a
     expect(clip.infer.to_s).to eq('Foo, Bar')
   end
 
+  # pending https://github.com/castwide/solargraph/pull/836
+  xit 'handles parallel type possibilities with #nil?' do
```

Comment:
I suspect there will also some untested related regressions with this PR solo - to be safe, I think we should release this PR in the same release as #836 .  I'll keep them as separate PRs for ease of review in the meantime.